src/sqlfluff/dialects/dialect_mysql.py

```diff
@@ -652,12 +652,24 @@ class DeleteStatementSegment(BaseSegment):
         OneOf(
             Sequence(
                 "FROM",
-                Delimited(Ref("TableReferenceSegment"), terminators=["USING"]),
+                Delimited(
+                    Sequence(
+                        Ref("TableReferenceSegment"),
+                        Sequence(Ref("DotSegment"), Ref("StarSegment"), optional=True),
+                    ),
```

Comment:
Can we make a tiny grammar for this and use it in the two places you have it here?